library/src/main/java/no/finntech/capturandro/BitmapUtil.java

```diff
@@ -52,7 +53,12 @@ static Uri getProcessedImage(Context context, InputStream inputStream, int longe
 
     @RequiresPermission(Manifest.permission.WRITE_EXTERNAL_STORAGE)
     static String getUniqueFilename(Context context) {
-        return new File(context.getExternalCacheDir().getAbsolutePath(), "capturandro-" + System.currentTimeMillis() + "." + random.nextInt() + ".jpg").toString();
+        if (HUAWEI_MANUFACTURER.equalsIgnoreCase(Build.MANUFACTURER) && Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
```

Comment:
We could refactor this a bit so that we avoided the duplication of the leaf-name part of the job.
Also, could we add a source-comment on what is going on here and ideally a link to a SO article or similar?